### ISSUE TITLE
After Ctrl + Alt + U，All show "Uploading..."

### ISSUE THREAD

**Issue by CatcherX** [PicGo/vs-picgo#34]
Version: 1.33.1 (user setup)
Commit: 51b0b28134d51361cf996d2f0a1c698247aeabd8
Date: 2019-04-11T08:27:14.102Z
Electron: 3.1.6
Chrome: 66.0.3359.181
Node.js: 10.2.0
V8: 6.6.346.32
OS: Windows_NT x64 6.1.7601

PicGo：2.0.3

**Comment by Spades-S** [comment]
@CatcherX  Upload successfully when select `sm.ms` as the image hoisting service. **Emmm.... maybe the network is very bad.** Can you add more info, such as settings...

![Kapture 2019-04-19 at 18 20 57](https://user-images.githubusercontent.com/18718461/56420336-1c5bab00-62d0-11e9-94d8-1ce3db9817f6.gif)

**Comment by CatcherX** [comment]
My network is normal. Uploading images from explorer is very fast.But uploading an image from clipboard is all show uploading, the log file not not recorded.

**Comment by upupming** [comment]
Please show your log in vscode. and seetings.json

**Comment by upupming** [comment]
@CatcherX Just a reminder, you can see your vscode console by click `Help -> Toggle Developer Tools` and you should see something like this:

![image](https://user-images.githubusercontent.com/24741764/56421261-3f885980-62d4-11e9-95ea-6aea7b2b82ef.png)

You `settings.json` can be opened by `File -> Preference -> Settings` or `Ctrl + ,`. 

Please copy error logged at the console and your `setttings.json` here.

@Spades-S We probably should update the issue template!

**Comment by CatcherX** [comment]
> @CatcherX Just a reminder, you can see your vscode console by click `Help -> Toggle Developer Tools` and you should see something like this:
> 
> ![image](https://user-images.githubusercontent.com/24741764/56421261-3f885980-62d4-11e9-95ea-6aea7b2b82ef.png)
> 
> You `settings.json` can be opened by `File -> Preference -> Settings` or `Ctrl + ,`.
> 
> Please copy error logged at the console and your `setttings.json` here.
> 
> @Spades-S We probably should update the issue template!

`ERR spawn powershell ENOENT: Error: spawn powershell ENOENT
	at Process.ChildProcess._handle.onexit (internal/child_process.js:229:19)
	at onErrorNT (internal/child_process.js:406:16)
	at process._tickCallback (internal/process/next_tick.js:63:19)`

**Comment by Molunerfinn** [comment]
@CatcherX Open your `cmd` and write `powershell`, enter，take a screenshot.

**Comment by Molunerfinn** [comment]
It may be the same issue as [this one](https://github.com/felixrieseberg/windows-build-tools/issues/20)

**Comment by Spades-S** [comment]
> @CatcherX Just a reminder, you can see your vscode console by click `Help -> Toggle Developer Tools` and you should see something like this:
> 
> ![image](https://user-images.githubusercontent.com/24741764/56421261-3f885980-62d4-11e9-95ea-6aea7b2b82ef.png)
> 
> You `settings.json` can be opened by `File -> Preference -> Settings` or `Ctrl + ,`.
> 
> Please copy error logged at the console and your `setttings.json` here.
> 
> @Spades-S We probably should update the issue template!

Yeah, urgent!

**Comment by CatcherX** [comment]
> It may be the same issue as [this one](https://github.com/felixrieseberg/windows-build-tools/issues/20), checkout if `powershell` is in your system PATH.
> 
> ![image](https://user-images.githubusercontent.com/12621342/56422028-0baf3300-62d8-11e9-8140-0cd01a20e113.png)

After put the powershell in system PATH，try again，show:

![2019-4-19-19-28-25_11.jpg](https://i.loli.net/2019/04/19/5cb9b0de03a6c.jpg)

**Comment by CatcherX** [comment]
`{
    "workbench.colorTheme": "Visual Studio Dark",
    "git.autofetch": true,
    "picgo.picBed.github.path": "/",
    "picgo.customUploadName": "${dateTime}_${fileName}${extName}",
    "picgo.configPath": "D:\\PicGo\\bed.json",
    "http.proxy": ""
}`

**Comment by Molunerfinn** [comment]
What's your system path? take shot

**Comment by CatcherX** [comment]
![2019-4-19-19-44-2_path.jpg](https://i.loli.net/2019/04/19/5cb9b4862c202.jpg)

**Comment by Spades-S** [comment]
@CatcherX  The config path is WRONG!
![image](https://user-images.githubusercontent.com/18718461/56422917-5af76280-62dc-11e9-84db-87cd5d9a0bf6.png)
The right one is `D:\\Picgo\\bed.json`

**Comment by CatcherX** [comment]
![2019-4-19-19-52-40_cccc.jpg](https://i.loli.net/2019/04/19/5cb9b68c10165.jpg)
> The right is `D:\\Picgo\\bed.json`

I am sorry.That is after escape,Actually, it's right.

**Comment by Molunerfinn** [comment]
If you are using `Win7`, you can just upload clipboard with image content, not the image file itself.

如果你在使用`Win7`，由于powershell版本的限制，你要上传剪贴板图片是有条件的。比如你不能复制一个「图片文件」然后上传，你只能复制「图片内容」，这样才能通过PicGo的剪贴板上传。什么叫「图片内容」？比如你用QQ截图，这个时候截图是以图片内容的形式存放在剪贴板里；你在网页上右键一张图片，选择「复制图片」的时候，是以图片内容的形式存放在剪贴板里。但是你要是在系统里右键一张图片文件，然后选择复制，这样是以「图片文件」的形式存放在剪贴板里的，PicGo在Win7的里是不支持这种形式上传的。

**Comment by Molunerfinn** [comment]
You can try this:

1. reboot your PC
2. open VSCode
3. take a screenshot
4. upload with vs-picgo by `uploading with clipboard`

Let's see what will happen

**Comment by CatcherX** [comment]
![2019-4-19-20-9-30_888.jpg](https://i.loli.net/2019/04/19/5cb9ba7ca85b5.jpg)
![2019-4-19-20-10-26_111.jpg](https://i.loli.net/2019/04/19/5cb9bab4db494.jpg)

是截图上传的，在我的配置文件目录，能看到从粘贴板传进来的文件

**Comment by Molunerfinn** [comment]
嗯大概知道是哪里的问题了。不过没有Win7的系统没有办法深入测试...

**Comment by Molunerfinn** [comment]
目前的问题大概是PicGo从剪贴板里通过Powershell生成了图片之后，powershell返回的图片路径为空，导致上传的时候传给PicGo的路径是`''`，就上传失败。

**Comment by Molunerfinn** [comment]
If you use vs-picgo's default config instead of yourself config file, what will happen

**Comment by upupming** [comment]
> _{_
> "workbench.colorTheme": "Visual Studio Dark",
> "git.autofetch": true,
> "picgo.picBed.github.path": "/",
> "picgo.customUploadName": "${dateTime}_${fileName}${extName}",
> "picgo.configPath": "D:\PicGo\bed.json",
> "http.proxy": ""
> }
> 
> ![2019-4-19-19-33-31_config.jpg](https://camo.githubusercontent.com/7bb9d606aafbc84259a014becdae50c7dc93a584/68747470733a2f2f692e6c6f6c692e6e65742f323031392f30342f31392f356362396232306632653537662e6a7067)

You are mixing the style of the configuration of v1.x and v2.0. Please delete all the `picgo` and `picBed` object, and only keep something like these is okay:

```json
// ...
"picgo.configPath": "C:\\Users\\Doraeming\\AppData\\Roaming\\PicGo\\data.json",
  "picgo.customUploadName": "${fileName}-${dateTime}",
  "picgo.customOutputFormat": "<img src=\"${url}\" alt=\"${uploadedName}\">",
// ...
```

**Comment by Molunerfinn** [comment]
@CatcherX `ctrl+alt+o` is the `upload from the input box`, you should use `ctrl+alt+u` to upload the clipboard files!

![image](https://user-images.githubusercontent.com/12621342/56424477-00153980-62e3-11e9-8dee-a4c0f0e627e8.png)

**Comment by CatcherX** [comment]
@upupming 
![2.jpg](https://i.loli.net/2019/04/19/5cb9cb8ec7a8a.jpg)
![222.jpg](https://i.loli.net/2019/04/19/5cb9cb8de2978.jpg)
![333333333.jpg](https://i.loli.net/2019/04/19/5cb9cb950847b.jpg)

I'm trying again on another computer.Use the default settings.
The mistake is still like this.

**Comment by upupming** [comment]
__

**Comment by Molunerfinn** [comment]
PicGo-Core [v1.3.6](https://github.com/PicGo/PicGo-Core/commit/48b72ed85b83d20d222eaf3ebed0fa9a41fb704e) fix this bug

**Comment by Spades-S** [comment]
@CatcherX  Fixed in vs-picgo 2.0.4. Please upgrade vs-picgo
